Source/Core/Core/FifoPlayer/FifoRecordAnalyzer.cpp

```diff
@@ -1,104 +0,0 @@
-// Copyright 2011 Dolphin Emulator Project
-// Licensed under GPLv2+
-// Refer to the license.txt file included.
-
-#include "Core/FifoPlayer/FifoRecordAnalyzer.h"
-
-#include <algorithm>
-
-#include "Common/MsgHandler.h"
-#include "Core/FifoPlayer/FifoAnalyzer.h"
-#include "Core/FifoPlayer/FifoRecorder.h"
-#include "Core/HW/Memmap.h"
-
-using namespace FifoAnalyzer;
-
-void FifoRecordAnalyzer::Initialize(const u32* cpMem)
-{
-  s_DrawingObject = false;
-
-  FifoAnalyzer::LoadCPReg(VCD_LO, cpMem[VCD_LO], s_CpMem);
-  FifoAnalyzer::LoadCPReg(VCD_HI, cpMem[VCD_HI], s_CpMem);
-  for (u32 i = 0; i < CP_NUM_VAT_REG; ++i)
-    FifoAnalyzer::LoadCPReg(CP_VAT_REG_A + i, cpMem[CP_VAT_REG_A + i], s_CpMem);
-
-  const u32* const bases_start = cpMem + ARRAY_BASE;
-  const u32* const bases_end = bases_start + s_CpMem.arrayBases.size();
-  std::copy(bases_start, bases_end, s_CpMem.arrayBases.begin());
-
-  const u32* const strides_start = cpMem + ARRAY_STRIDE;
-  const u32* const strides_end = strides_start + s_CpMem.arrayStrides.size();
-  std::copy(strides_start, strides_end, s_CpMem.arrayStrides.begin());
-}
```

Comment:
Note that this code failed to load `CP_VAT_REG_B` and `CP_VAT_REG_C`, which caused errors when recording for some games where the initial CP state matters (for instance, Need for Speed: Most Wanted was affected).  This has been fixed with the new code, which also made this kind of loading happen in only one place.